src/puppetlabs/puppetdb/client.clj

```diff
@@ -41,10 +40,12 @@
     payload :- {s/Any s/Any}
     timeout]
    (let [body (json/generate-string payload)
+         ;; is it ok to not account for benchmark sending string keys when everyting else uses :producer_timestamp???
+         producer_timestamp (-> payload :producer_timestamp str)
```

Comment:
Getting "" here for producer_timestamp with benchmark might not be what we want.

Reply:
```
   (let [body (json/generate-string payload)
         ;; is it ok to not account for benchmark sending string keys when everyting else uses :producer_timestamp???
         producer_timestamp (-> payload :producer_timestamp)
         url_params (str
                      (format "?command=%s&version=%s&certname=%s"
                        (str/replace command #" " "_") version certname)
                      (when producer_timestamp
                        (format "&producer_timestamp=%s" producer_timestamp)))
         url (str (utils/base-url->str base-url)
                  url_params
                  (when timeout (format "&secondsToWaitForCompletion=%s" timeout)))]
```

Something like that will prevent benchmark from triggering bash in place but other optimizations still make benchmark kinda useless as a load testing tool.